src/metamodels/generic.py

```diff
@@ -0,0 +1,127 @@
+'''Madeleine proposes that she write a GenericMetamodel class that implements
+IBayesDBMetamodel. This class will wrap other models, including gpmcc and
+Crosscat. It will replace src/metamodels/crosscat.py and the gpmcc_metamodel
+package.
+
+When a generator is initialized, it will create a ModelState for each model.
+When a generator is analyzed, it will call the transition_* methods on the
+models. When BQL-relevant calls are made, it will translate those to GPM calls
+and call the appropriate methods on the model property of the GPM state.
+
+The main purposes of this project are:
+    (1) Move all metamodel SQL code into a single module. (Serialization will
+        not be backward compatible; existing bdbs will use CrosscatMetamodel.)
+    (2) Reduce the amount of bayeslite-specific code necessary to implement a
+        new model to just the GPM interface. For example, GenericMetamodel will
+        have a predict_confidence method that is implemented in terms of
+        simulate.
+    (3) Allow us to write alternative implementations of GenericMetamodel that
+        are remote or distributed.
+
+Some open issues:
+    (1) Madeleine is skeptical of the utility of rename_column and insertmany,
+        which allow mutation of generators. She is likely to have them throw
```

Comment:
Agree on `rename_column`, but what are your qualms with `insertmany`? In fact, we also need `remove_many`.

A GPM is a dynamic/algorithmic interpretation of traditional "statistical" models from R, and is required to support incremental incorporate/unincorporate, even if the underlying model implementation needs itself to "retrain" all over again (i.e. re-run a regression analysis) before/after each observation. My hunch, however, is that most interesting probabilistic models we express as GPMs will not have this inefficient property.